lib/gateway/Shard.js

```diff
@@ -638,11 +638,12 @@ class Shard extends EventEmitter {
                 if(packet.d.user.username !== undefined) {
                     let user = this.client.users.get(packet.d.user.id);
                     let oldUser = null;
-                    if(user && (user.username !== packet.d.user.username || user.discriminator !== packet.d.user.discriminator || user.avatar !== packet.d.user.avatar)) {
+                    if(user && (user.username !== packet.d.user.username || user.discriminator !== packet.d.user.discriminator || user.avatar !== packet.d.user.avatar || user.avatar_decoration !== packet.d.user.avatar_decoration)) {
                         oldUser = {
                             username: user.username,
                             discriminator: user.discriminator,
-                            avatar: user.avatar
+                            avatar: user.avatar,
+                            avatar_decoration: user.avatar_decoration
```

Comment:
The `user` variable is a User object:

```suggestion
                            avatarDecoration: user.avatarDecoration
```

Reply:
The `PartialUser` type defines `avatar_decoration` already, wouldn't it be something like this?

```suggestion
                            avatar_decoration: user.avatarDecoration
```

Reply:
Having taken a look at the `PartialUser` type, a majority of its usage seems to be with raw API data, leaving the `userUpdate` definition as being the odd one out. FWIW it'd be more worth it to use a separate `OldUser` type for this to keep things consistent with the rest of the library (with `avatarDecoration` camel-cased)